.pre-commit-config.yaml

```
@@ -694,10 +694,11 @@ repos:
       - id: check-base-operator-usage
         language: pygrep
         name: Check BaseOperatorLink core imports
-        description: Make sure BaseOperatorLink is imported from airflow.models.baseoperatorlink in core
-        entry: "from airflow\\.models import.* BaseOperatorLink"
+        description: Make sure BaseOperatorLink is imported from airflow.sdk in core
+        entry: ".*from airflow\\.sdk\\..* import.* BaseOperatorLink\\b"
```

Having the `\..*` part on the end somewhat defies the point of the check.

This check existed to ensure that it was done as `from airflow.models import BaseOperatorLink` and not `from airflow.models.baseoperator import BaseOperatorLink` -- so I think this should be

```suggestion
        description: Make sure BaseOperatorLink is imported from airflow.sdk in core
        entry: "[ ]*from airflow\\.sdk import.* BaseOperatorLink\\b"
```

---

Yeah lets do this for now, but as you mentioned below we should ideally create a discussion or think about how we do imports in core.